contrib/confluentinc/confluent-kafka-go/kafka.v2/types.go

```diff
@@ -130,6 +130,21 @@ func (w wEvent) KafkaOffsetsCommitted() (kafkatrace.OffsetsCommitted, bool) {
 	return nil, false
 }
 
+type wTopicPartitionError struct {
+	error
+}
+
+func (w wTopicPartitionError) IsGenericServerError() bool {
+	if w.error == nil {
+		return false
+	}
+	return w.error.(kafka.Error).Code() == kafka.ErrUnknown
```

Comment:
unsure if this type assertion could fail, but just in case it is better to check it (also it is better to use `errors.As` for type asserting errors):

```
var kafkaErr kafka.Error
if errors.As(err, &kafkaErr) {
	...
}
```